### PR TITLE
Much improved batch evolve (aka evolve all) mechanism

### PR DESCRIPTION
**Pull request by guyz**
Short Description: 

Selective batch evolve on startup. See improvements below:

Fixes:
- Pokemons are evolved from high CP to low CP.
- Can specify "all" to evolve the entire inventory or a list (e.g., "Pidgey,Weedle") to only evolve specific types.
- Evolved pokemons are automatically transferred based on the release criteria.

TODO: release and transfer code was copied from catch worker. We need to refactor these functionalities out of the workers and into utils so that they can easily re-use this logic.

### CODE REVIEW

**Comment by michi0987** [comment]
Works for me good job :)

**Comment by krook1024** [comment]
Good job, @guyz.

@michi0987 How does it make sense to evolve a pokemon and then transfer it?

**Comment by Witzbold** [comment]
works fine for me and it is very fast. 
I didn't get it stopped with ctrl + c though.

File: pokemongo_bot/cell_workers/evolve_all_worker.py [hunk 1]
```diff
@@ -20,34 +21,220 @@ def work(self):
...
+            print('[#] Successfully evolved {} with {} cp!'.format(
+                pokemon_name, pokemon_cp
             ))
         else:
             # cache pokemons we can't evolve. Less server calls
             cache[pokemon_name] = 1
+        sleep(5.7)
+
+    # TODO: move to utils. These methods are shared with other workers.
+    def transfer_pokemon(self, pid):
```
Comment (TheSavior) [inline pokemongo_bot/cell_workers/evolve_all_worker.py]: Can you move these out before this PR? It doesn't make sense to have duplicated code between the workers and this is 120 lines of copy and paste.

**Comment by solderzzc** [comment]
README need a update too. :) amazing.

**Comment by orgkhnargh** [comment]
@krook1024 

> How does it make sense to evolve a pokemon and then transfer it?

Evolving yields XP.

**Pull request merged**